### PR TITLE
Add test cases for `#sanitize_sql_array` with named_bind_variables

### PR DESCRIPTION
**Pull request by yui-knk**
And add code examples to `sanitize_sql_for_conditions`,
`sanitize_sql_for_assignment`, and `sanitize_sql_array`.

### CODE REVIEW

File: activerecord/test/cases/sanitize_test.rb [hunk 1]
```diff
@@ -25,6 +25,15 @@ def test_sanitize_sql_array_handles_bind_variables
     assert_equal "name=#{quoted_bambi_and_thumper}", Binary.send(:sanitize_sql_array, ["name=?", "Bambi\nand\nThumper".mb_chars])
   end
 
+  def test_sanitize_sql_array_handles_named_bind_variables
+    quoted_bambi = ActiveRecord::Base.connection.quote("Bambi")
+    assert_equal "name=#{quoted_bambi}", Binary.send(:sanitize_sql_array, ["name=:name", name: "Bambi"])
+    assert_equal "name=#{quoted_bambi}", Binary.send(:sanitize_sql_array, ["name=:name", name: "Bambi".mb_chars])
```
Comment (sgrif) [inline activerecord/test/cases/sanitize_test.rb]: Why do we care about `mb_chars` here? Seems like that has nothing to do with testing that named bind variables are handled.

**Comment by yui-knk** [comment]
@sgrif Thanks I agree we do not need to care about `mb_chars`! I will remove it.
`mb_chars` appeared by [this commit](https://github.com/rails/rails/commit/1585a7ed0228c2d4688e986c24fc221a4a9e5104) and never be removed to today :cry: .
BTW I think we can remove `mb_chars` from other tests in sanitize_test.rb (ex: test_sanitize_sql_array_handles_string_interpolation), I will send another PR!

**Comment by sgrif** [comment]
Thanks. Can we have the test case also test using more than one variable, and using a variable with the same name more than once?

**Comment by yui-knk** [comment]
Do you mean?

> test using more than one variable

```ruby
Binary.send(:sanitize_sql_array, ["name=:name AND group_id=:group_id", name: "Bambi", group_id: 1])
```

> using a variable with the same name more than once

```ruby
Binary.send(:sanitize_sql_array, ["name=:name", name: "Bambi", name: "Bambi\nand\nThumper"])
```

If so, I want to change `Binary` to `Author`, because `Author` has `name` and `organization_id` (in this test real SQL is not generated but I do not like to pass invalid sql to `sanitize_sql_array`)

```ruby
  create_table :authors, force: true do |t|
    t.string :name, null: false
    t.integer :author_address_id
    t.integer :author_address_extra_id
    t.string :organization_id
    t.string :owned_essay_id
  end
```

**Comment by yui-knk** [comment]
ref #22209

**Comment by yui-knk** [comment]
@sgrif Updated!

**Pull request merged**